File: server/routes/templates.js
import express from 'express';
const router = express.Router();
import { auth } from '../middleware/auth.js';
import TaskTemplate from '../models/TaskTemplate.js';

router.get('/', auth, async (req, res) => {
  try {
    const templates = await TaskTemplate.find()
      .populate('createdBy', 'name email')
      .sort({ createdAt: -1 });

    res.json(templates.map(t => ({
      id: t._id.toString(),
      name: t.name,
      title: t.title,
      description: t.description,
      tags: t.tags,
      priority: t.priority,
      estimatedHours: t.estimatedHours,
      createdBy: t.createdBy._id.toString(),
      createdAt: t.createdAt.toISOString(),
    })));
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

router.post('/', auth, async (req, res) => {
  try {
    const template = new TaskTemplate({
      ...req.body,
      createdBy: req.user._id,
    });

    await template.save();

    res.status(201).json({
      id: template._id.toString(),
      name: template.name,
      title: template.title,
      description: template.description,
      tags: template.tags,
      priority: template.priority,
      estimatedHours: template.estimatedHours,
      createdBy: template.createdBy.toString(),
      createdAt: template.createdAt.toISOString(),
    });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

router.delete('/:id', auth, async (req, res) => {
  try {
    await TaskTemplate.findByIdAndDelete(req.params.id);
    res.json({ success: true });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

export default router;

File: backend/app/routes/templates.py
from fastapi import APIRouter, Depends, HTTPException, status
from app.auth import get_current_user, transform_record
from app.database import get_admin_client
from app.schemas.task_template import TemplateCreate

router = APIRouter()


@router.get("")
async def get_templates(current_user: dict = Depends(get_current_user)):
    supabase = get_admin_client()
    response = (
        supabase.table("task_templates")
        .select("*")
        .order("created_at", desc=True)
        .execute()
    )
    return [transform_record(t) for t in response.data]


@router.post("")
async def create_template(
    body: TemplateCreate, current_user: dict = Depends(get_current_user)
):
    supabase = get_admin_client()
    template_data = {
        "name": body.name,
        "title": body.title,
        "description": body.description or "",
        "tags": body.tags or [],
        "priority": body.priority or "medium",
        "estimated_hours": body.estimatedHours or 0,
        "created_by": current_user["id"],
    }

    response = supabase.table("task_templates").insert(template_data).execute()
    return transform_record(response.data[0])


@router.delete("/{template_id}")
async def delete_template(
    template_id: str, current_user: dict = Depends(get_current_user)
):
    supabase = get_admin_client()
    supabase.table("task_templates").delete().eq("id", template_id).execute()
    return {"success": True}
